rllib/agents/sac/sac_policy.py

```diff
@@ -233,11 +234,16 @@ def sac_actor_critic_loss(policy, model, _, train_batch):
                 weights=0.5))
 
     # Auto-calculate the target entropy.
-    if policy.config["target_entropy"] == "auto":
+    if policy.config["target_entropy"] == "auto" or \
+            policy.config["target_entropy"] is None:
+        # See hyperparams in [2] (README.md).
         if model.discrete:
-            target_entropy = np.array(-policy.action_space.n, dtype=np.float32)
+            target_entropy = 0.98 * np.array(
```

Comment:
@michaelzhiluo @hartikainen

Reply:
What is the 0.98 for?

Reply:
No sure either. I don't think it matters much. It's from the SAC-discrete paper. The original SAC paper only has the auto-target-H calculation for cont spaces.

Reply:
https://arxiv.org/abs/1910.07207
Appendix

Reply:
I don't have experience with discrete SAC so don't have a direct answer to what the right number here would be. The way I would determine the value is to 1) see what the noise values for similar algorithms are and try to roughly calculate what those would correspond to in terms of entropy (I *think* this should be doable with categorical distribution), and 2) run a few environments, maybe sweeping around the values determined in (1), and see what values work the best.

But if the paper above converged to the value used here by some reasonable heuristic, I guess it would be okay to use that one 🙂